api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -985,4 +985,12 @@ public DbUser confirmProfile() {
     dbUser.setProfileLastConfirmedTime(new Timestamp(clock.instant().toEpochMilli()));
     return userDao.save(dbUser);
   }
+
+  /** Confirm that a user has either reported any AoU-related publications, or has none. */
+  @Override
+  public DbUser confirmPublications() {
+    final DbUser dbUser = userProvider.get();
+    dbUser.setPublicationsLastConfirmedTime(new Timestamp(clock.instant().toEpochMilli()));
+    return userDao.save(dbUser);
```

Comment:
nit: I prefer to return void in setter functions like this in interest of the single responsibility principle. in this case, you can use this function either 1. set the publication time 2. retrieve the current request user.

Reply:
I prefer the opposite, for fluency reasons.  Since we have this information, we can return it for "free" because it will be useful for some callers.

Is your concern that a developer will see this method and use it as a way to retrieve the user?  That would not be desirable, since it would send a false compliance signal.

Reply:
Hmm, more so that I feel like it's potentially easier for bugs to slip into the implementation if its being used for two things.

For example, you could actually just change construct a new user object here or fetch an entirely different user and your tests would still pass since it's checking that the returned object is being modified and not the current user.

Reply:
good point - updated the test so it doesn't rely on this returned value, but instead checks the current user